cmd/gnotxport/export.go

```diff
@@ -10,45 +12,99 @@ import (
 
 	"github.com/gnolang/gno/pkgs/amino"
 	"github.com/gnolang/gno/pkgs/bft/rpc/client"
-	"github.com/gnolang/gno/pkgs/command"
+	"github.com/gnolang/gno/pkgs/commands"
 	"github.com/gnolang/gno/pkgs/std"
-
 	// XXX better way?
 	_ "github.com/gnolang/gno/pkgs/sdk/auth"
 	_ "github.com/gnolang/gno/pkgs/sdk/bank"
 	_ "github.com/gnolang/gno/pkgs/sdk/vm"
 )
 
-type txExportOptions struct {
-	Remote      string `flag:"remote" help:"Remote RPC addr:port"`
-	StartHeight int64  `flag:"start" help:"Start height"`
-	TailHeight  int64  `flag:"tail" help:"Start at LAST - N"`
-	EndHeight   int64  `flag:"end" help:"End height (optional)"`
-	OutFile     string `flag:"out" help:"Output file path"`
-	Quiet       bool   `flag:"quiet" help:"Quiet mode"`
-	Follow      bool   `flag:"follow" help:"Keep attached and follow new events"`
+type exportCfg struct {
```

Comment:
not sure about using the same struct for opts and the `interface{ Exec(...) }`.

having a standalone `opts` allows better composability and to declare opts and flags from a pkg, while keeping the command from the main.

what do you think?

Reply:
I actually agree about separating out the flag holder logic (the configuration) and the actual run method for the command. The way I've set it up now follows the guidelines set out in `ffcli`'s examples.

The tricky part here is to conform to the [Exec method signature](https://github.com/peterbourgon/ff/blob/c40b025c3985f9fcb13832a3b245f6e1a95d921f/ffcli/command.go#L80), because we want to access the configuration itself inside the Exec method.
The only possible solutions I see here are:
- define `Exec` as a member method of the configuration (we have this now)
- define the `Exec` method in the same scope that we are creating the command, so the configuration is in the same scope:
```go
cfg := &Configuration{}

// ...

cmd := NewCommand(..., func(ctx context.Context, args []string) { // implement command... })

````
- and finally, pass in the configuration through the `context.Value`. I've done this in an alternate branch commit, which you can check out [here](https://github.com/zivkovicmilos/gno/commit/e34b0b527f9d15382b1c3624af826aceb2c0957f)

If we don't need the `args`, as they're optional, we can easily have `NewCommand` take in a `func(cfg Config) error`, and set it to our execution method that just takes in a specific configuration. Additionally, if we need the context (for actions like backup / restore, sigkills etc), we can require the `Config` interface to have a `GetContext() context.Context` getter or setter method. 

Let me know what you think is the best way to go, whether we should keep the execution method a free function, or make it a member method like we do now

Reply:
Resolved this comment in a later note:
https://github.com/gnolang/gno/pull/497#issuecomment-1422317925